app/controllers/api/subcollections/alert_statuses.rb

```diff
@@ -0,0 +1,25 @@
+module Api
+  module Subcollections
+    module AlertStatuses
+      include Subcollections::AlertStatusStates
+
+      def alert_statuses_query_resource(object)
+        payload = {}
+        payload = {"environment" => "production"}.merge("alerts" => alerts_and_states(object)) if object.kind_of?(ManageIQ::Providers::CloudManager) ||
```

Comment:
This seems like the wrong place to define which providers support this behavior. Could it be pushed down to the model layer?